src/diagnosticMessages.json

```diff
@@ -56,6 +56,8 @@
   "Indexed access may involve bounds checking.": 904,
   "Explicitly returning constructor drops 'this' allocation.": 905,
   "Unnecessary definite assignment.": 906,
+  "Comparision with NaN is invariant. You need to use isNaN(x).": 907,
```

Comment:
Perhaps
```suggestion
  "Comparison with NaN is always false. Use isNaN(x) instead.": 907,
```

Reply:
> "Comparison with NaN is always false

That's not precise. `x == NaN` always false. `x != NaN` always true

Reply:
```suggestion
  "NaN does not compare equal to any other value including itself. Use isNaN(x) instead.": 907,
```